### PR TITLE
Switch CI to GitHub Actions

### LINKED ISSUE #243: Migrating to GitHub Actions
I am considering to migrate the CI to GitHub Actions from Travis CI, due to the Travis CI's announcement.
https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

### PR DESCRIPTION
**Pull request by nforro**
I had to remove some test scenarios because I just couldn't get them to work, for example there is no Python 2 in Fedora anymore,  CentOS 6 repos don't seem to be accessible, etc.

Fixes #243.

### CODE REVIEW

**Comment by nforro** [comment]
Link to the latest CI status: https://github.com/nforro/rpm-py-installer/actions/runs/2657797843

**Comment by junaruga** [comment]
Thank you so much for this PR. I will take a look!

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Build test images and run the tests
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 20 * * 2'
```
[the remark this answers is not in the document]
Comment (nforro) [reply .github/workflows/build-and-test.yml:6]: Oh, ok, from the Travis CI logs it seemed to me it used to be scheduled for Tuesday.

[the remark this answers is not in the document]
Comment (junaruga) [reply .github/workflows/build-and-test.yml:6]: You are right. sorry for my mistake. The "2" is not Wednesday but Tuesday.
https://www.tutorialspoint.com/unix_commands/crontab.htm

**Comment by junaruga** [comment]
I am still reviewing. I will comment one by one if it is needed. The PR looks excellent!

File: .github/workflows/build-and-test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (junaruga) [reply .github/workflows/build-and-test.yml:6]: >  from the Travis CI logs it seemed to me it used to be scheduled for Tuesday.

Yes, you are right. Maybe I had been executing the cron every Tuesday 20:00 seeing the logs.
https://travis-ci.org/github/junaruga/rpm-py-installer/builds

**Comment by junaruga** [comment]
I understand this PR is important, and also related to another issue ticket https://github.com/junaruga/rpm-py-installer/issues/155 . This week I will spend time reviewing this PR. Thanks for your patience.

**Comment by nforro** [comment]
That's fine. I'm not sure this PR solves the MacOS installation issue, but it would be a nice bonus :slightly_smiling_face: 
BTW, if you need help maintaining this project, I think I might be able to give you a hand.

**Comment by junaruga** [comment]
> BTW, if you need help maintaining this project, I think I might be able to give you a hand.

Thanks! Today I couldn't review it. Sigh.. And thanks for the offer. I need help maintaining this project. It would be fantastic if you join as a maintainer of this project🤩  I sent an invitation to you to be the collaborator!

**Comment by junaruga** [comment]
> I had to remove some test scenarios because I just couldn't get them to work, for example there is no Python 2 in Fedora anymore, CentOS 6 repos don't seem to be accessible, etc.

Thanks, I reviewed! Some notes for us. Those are okay. I am going to merge this PR.

* The 3 cases `DOCKER="docker"`,  ``DOCKER="podman"` or No `DOCKER` before this PR were now tested as 2 cases `"docker": "docker"` or No `docker` key in the json file `.github/workflows/matrix_includes.json`.
  ```
  -    - name: fedora_32
  -      env: DOCKER="podman" IMAGE="fedora:32" TOXENV="py37,py27-cov"
  -      script: make test-no-network
  ```
* The `py26` case was dropped as the CentOS6 case was dropped due to the inaccessible package repository.
* Each comments for each matrix case was dropped, as the JSON file `.github/workflows/matrix_includes.json` used to manage the matrix cases doesn't support the comment feature. To manage the comments in a JSON file, there is [a tricky tip](https://www.freecodecamp.org/news/json-comment-example-how-to-comment-in-json-files/). But we don't do this.

  The branch names conditions to trigger CI was drooped.
  ```
  -branches:
  -  except:
  -    # Enable branches except the working in progress branch or
  -    # current pull-request feature/ci-centos.
  -    - /^(wip\/|feature\/ci-centos5)/
  ```
  In the new settings, any branch names can trigger CI in both push and pull-request.
  ```
  +on:
  +  push:
  +  pull_request:
  ```

**Comment by junaruga** [comment]
Oh, there are 2 test failures on OpenSuse Tumbleweed case.
https://github.com/junaruga/rpm-py-installer/runs/7565666546?check_suite_focus=true#step:10:163

**Comment by nforro** [comment]
> Oh, I got 2 test failures on OpenSuse Tumbleweed case. https://github.com/junaruga/rpm-py-installer/runs/7565666546?check_suite_focus=true#step:10:163

Ok, something must have changed since I last run the tests, let me take a look.

**Comment by junaruga** [comment]
Ok, thanks!

**Comment by nforro** [comment]
Fixed in https://github.com/junaruga/rpm-py-installer/pull/246.

**Pull request merged**